onnxruntime/core/session/inference_session.cc

```diff
@@ -97,41 +98,175 @@ inline std::basic_string<T> GetCurrentTimeString() {
 
 std::atomic<uint32_t> InferenceSession::global_session_id_{1};
 
-InferenceSession::InferenceSession(const SessionOptions& session_options,
-                                   logging::LoggingManager* logging_manager)
-    : session_options_(session_options),
-      graph_transformation_mgr_(session_options.max_num_graph_transformation_steps),
-      logging_manager_(logging_manager),
+static Status FinalizeSessionOptions(const SessionOptions& user_provided_session_options,
+                                     const ONNX_NAMESPACE::ModelProto* model_proto,
+                                     /*out*/ SessionOptions& finalized_session_options) {
+  const logging::Logger& default_logger = logging::LoggingManager::DefaultLogger();
+
+  // By now the environment should have initialized. (It is enforced prior to this.)
+  const Env& env_instance = Env::Default();
+
+  bool session_options_from_model = false;
+
+  // Get the value held by the environment variable - kOrtLoadConfigFromModelEnvVar
+  const std::string& load_config_from_model_env_var_value =
+      env_instance.GetEnvironmentVar(inference_session_utils::kOrtLoadConfigFromModelEnvVar);
```

Comment:
You can just use std::string instead of a const & as that's what GetEnvironmentVar returns. RVO will mean there's no extra copy (not that it matters here).

Reply:
yes, just pushed that commit. RVO should take care of it.